agent/engine/docker_task_engine.go

```diff
@@ -741,16 +776,23 @@ func (engine *DockerTaskEngine) pullAndUpdateContainerReference(task *api.Task,
 	if container.IsInternal() {
 		return metadata
 	}
+	PullSucceeded := metadata.Error == nil
+	engine.updateContainerReference(PullSucceeded, container, task.Arn)
```

Comment:
Can this logic be handled here - https://github.com/aws/amazon-ecs-agent/blob/159ae5c3067e0a0d7387ff68af2a28bc90f3baaa/agent/engine/docker_image_manager.go#L112 
i.e, depending on `InspectImage`, set `PullSucceeded` value?

Reply:
I think inspect image and pull image are two different things, when you inspect an image, the image can be there because it' reconfigured with the AMI, instead of being pulled from user's ECR, this matters when the setting is `once`.